saos-persistence/src/main/java/pl/edu/icm/saos/persistence/service/LawJournalEntryCodeExtractor.java

```diff
@@ -30,6 +30,10 @@ public int extractEntry(String lawJournalEntryCode) {
     //------------------------ PRIVATE --------------------------
     
     private int extractEntryCodePart(String lawJournalEntryCode, int partNumber) {
-        return Integer.parseInt(StringUtils.split(lawJournalEntryCode, LawJournalEntry.ENTRY_CODE_PARTS_SEPARATOR)[partNumber]);
+        try {
+            return Integer.parseInt(StringUtils.split(lawJournalEntryCode, LawJournalEntry.ENTRY_CODE_PARTS_SEPARATOR)[partNumber]);
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            throw new IllegalArgumentException(lawJournalEntryCode + " is invalid law journal entry code");
```

Comment:
It would be better to have a method checking correctness of the lawJournalEntryCode that could be used here and outside the class.